src/edu/ucsb/nceas/metacat/admin/MetacatAdminServlet.java

```diff
@@ -89,147 +108,143 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
+                             "**********************\n");
+
+        logMetacat.debug("\n****** START REQUEST PARAMETERS **********************\n"
+                             + RequestUtil.getParametersAsString(request)
+                             + "\n****** END REQUEST PARAMETERS   **********************\n");
 
+        try {
             if (!ConfigurationUtil.isBackupDirConfigured()) {
                 // if the backup dir has not been configured, then show the
                 // backup directory configuration screen.
                 processingMessage.add("You must configure the backup directory"
-                        + " before you can continue with Metacat configuration.");
+                                          + " before you can continue with Metacat configuration.");
                 RequestUtil.setRequestMessage(request, processingMessage);
                 action = "backup";
-                logMetacat.debug("MetacatAdminServlet.handleGetOrPost - "
-                                        + " Admin action changed to 'backup'");
+                logMetacat.debug(
+                    "MetacatAdminServlet - Admin action changed to 'backup'");
             } else if (!AuthUtil.isAuthConfigured()) {
                 // if authentication isn't configured, change the action to auth.
                 // Authentication needs to be set up before we do anything else
                 processingMessage.add("You must configure authentication before "
-                        + "you can continue with MetaCat configuration.");
+                                          + "you can continue with MetaCat configuration.");
                 RequestUtil.setRequestMessage(request, processingMessage);
                 action = "auth";
-                logMetacat.debug("MetacatAdminServlet.handleGetOrPost - "
-                                                            + "Admin action changed to 'auth'");
-            } else if (!AuthUtil.isUserLoggedInAsAdmin(request)) {
-                // If auth is configured, see if the user is logged in
-                // as an administrator.  If not, they need to log in before
-                // they can continue with configuration.
-                processingMessage.add("You must log in as an administrative "
-                                 + "user before you can continue with Metacat configuration.");
-                RequestUtil.setRequestMessage(request, processingMessage);
-                action = "login";
-                logMetacat.debug("MetacatAdminServlet.handleGetOrPost - "
-                                                    + "Admin action changed to 'login'");
+                logMetacat.debug(
+                    "MetacatAdminServlet - Admin action changed to 'auth'");
+            } else if (LoginAdmin.getInstance().needsLoginAdminHandling(request, action)) {
+                logMetacat.debug("MetacatAdminServlet - Admin action is: " + ACTION_LOGOUT
```

Comment:
Why the action is ACTTION_LOGOUT? To me, it can have lots possible values.

Reply:
good catch - thank you!

Reply:
fixed dc902054f1ad4c9cb5a9c0b3ccbb9c129f178525